crates/sdk/src/lib.rs

```diff
@@ -174,31 +174,41 @@ impl ProverClient {
     /// ### Examples
     ///
     /// ```no_run
+    /// use std::env;
+    /// use zkm_sdk::network::NetworkClientCfg;
     /// use zkm_sdk::ProverClient;
     ///
-    /// let private_key = std::env::var("ZKM_PRIVATE_KEY").unwrap();
-    /// let rpc_url = std::env::var("PROVER_NETWORK_RPC").ok();
-    /// let skip_simulation =
-    ///     std::env::var("SKIP_SIMULATION").map(|val| val == "true").unwrap_or_default();
-    ///
-    /// let client = ProverClient::network(private_key, rpc_url, skip_simulation);
+    ///  let endpoint = Some(env::var("ENDPOINT").unwrap_or("https://152.32.186.45:20002".to_string()));
+    ///  let domain_name = Some(env::var("DOMAIN_NAME").unwrap_or("stage".to_string()));
+    ///  let proof_network_privkey =
+    ///     Some(env::var("ZKM_PRIVATE_KEY").expect("ZKM_PRIVATE_KEY must be set for remote proving"));
+    ///  let current_dir = env::current_dir().expect("Failed to get current directory");
+    ///  let ca_cert_path = Some(current_dir.join("tool/ca.pem").to_string_lossy().to_string());
+    ///  let cert_path = Some(current_dir.join("tool/.pem").to_string_lossy().to_string());
+    ///  let key_path = Some(current_dir.join("tool/.key").to_string_lossy().to_string());
+    ///     
+    ///  let network_cfg = NetworkClientCfg {
+    ///     endpoint,
+    ///     ca_cert_path,
+    ///     cert_path,
+    ///     key_path,
+    ///     domain_name,
+    ///     proof_network_privkey,
+    ///  };
+    ///
+    ///  let client = ProverClient::network(&network_cfg);
     /// ```
-    #[cfg(any(feature = "network", feature = "network-v2"))]
-    pub fn network(_private_key: String, _rpc_url: Option<String>, _skip_simulation: bool) -> Self {
-        todo!();
-        // cfg_if! {
-        //     if #[cfg(feature = "network-v2")] {
-        //         Self {
-        //             prover: Box::new(NetworkProverV2::new(&private_key, rpc_url, skip_simulation)),
-        //         }
-        //     } else if #[cfg(feature = "network")] {
-        //         Self {
-        //             prover: Box::new(NetworkProverV1::new(&private_key, rpc_url, skip_simulation)),
-        //         }
-        //     } else {
-        //         panic!("network feature is not enabled")
-        //     }
-        // }
+    #[cfg(feature = "network")]
+    pub fn network(client_config: &NetworkClientCfg) -> Self {
```

Comment:
Remove the interface parameter and put the creation of NetworkClientCfg inside this method.